dap_aria_mapping/pipeline/app_tables/collaboration_network.py

```diff
@@ -0,0 +1,255 @@
+        #add assignees and the assignee types to patents tagged with topics
+        df = patents_with_topics_df.join(docs, how = "left", on = "publication_number")
+
+        #explode assignee name and type columns so each row contains a single assignee and the associated type vs a list
+        #NOTE: I DONT THINK THIS WORKS PROPERLY AS THE LISTS OF NAME TYPES SEEM TO BE OUT OF ORDER OF THE NAMES (see patent WO-2012141480-A2)
+        df = df.filter(
+            pl.col("assignee_harmonized_names").arr.lengths() == pl.col("assignee_harmonized_names_types").arr.lengths()
+            ).explode(
+                ["assignee_harmonized_names", "assignee_harmonized_names_types"]
+                )
+        #only consider organisations (not individuals)
+        orgs_df = df.filter(pl.col("assignee_harmonized_names_types") == "ORGANISATION")
+
+    #add area and domain col
+    orgs_df_with_topics = add_area_domain_chatgpt_names(
+        expand_topic_col(orgs_df).unique(subset=[id_col, "topic", org_name_col]))
+
+    print("GENERATING NETWORK")
+    network = nx.Graph()
+
+    print("Adding nodes with the following attributes: topics, areas, domains")
+    if args.top_n:
+        print("Only including top {} orgs".format(args.top_n))
+        node_metadata, nodes =  docs_per_org(orgs_df, id_col, org_name_col, top_n = args.top_n)
+        orgs_df = orgs_df.filter(pl.col(org_name_col).is_in(nodes))
+        orgs_df_with_topics = orgs_df_with_topics.filter(pl.col(org_name_col).is_in(nodes))
+        network.add_nodes_from(nodes)
+    else:
+        node_metadata = docs_per_org(orgs_df, id_col, org_name_col)
+        network.add_nodes_from(set(orgs_df[org_name_col]))
+
+    nx.set_node_attributes(network,node_metadata, "overall_doc_count")
+    nx.set_node_attributes(network, node_data(orgs_df_with_topics, "topic_name", org_name_col), "topics")
+    nx.set_node_attributes(network, node_data(orgs_df_with_topics, "area_name", org_name_col), "areas")
+    nx.set_node_attributes(network, node_data(orgs_df_with_topics, "domain_name", org_name_col), "domains")
+
+    print("Adding edges")
+    edges = get_edges(orgs_df, id_col, org_name_col)
+    network.add_edges_from(list(edges.keys()))
+    nx.set_edge_attributes(network, edges, "overall_doc_count")
+    nx.set_edge_attributes(network, edge_data(orgs_df_with_topics, by="topic_name", id_col= id_col, org_name_col = org_name_col), "topics")
+    nx.set_edge_attributes(network, edge_data(orgs_df_with_topics, by="area_name", id_col= id_col, org_name_col= org_name_col), "areas")
+    nx.set_edge_attributes(network, edge_data(orgs_df_with_topics, by="domain_name", id_col= id_col, org_name_col= org_name_col), "domains")
+
+    print("Saving network")
+    #upload_obj(network, BUCKET_NAME, "outputs/app_data/change_makers/networks/{}_{}.pkl".format(args.doc_type, args.level))
+    with open("outputs/academia_institutions.pickle", "wb") as f:
+        pickle.dump(network, f)
```

Comment:
it actually does let you do that! I had just commented it out because i was running it locally because my internet was timing out on the upload and forgot to comment it back in.